### ISSUE TITLE
`invalid_vector_dimensions` when using vector search

### ISSUE THREAD

**Issue by ahmednfwela** [meilisearch/meilisearch#4035]
Following the guide in https://blog.meilisearch.com/vector-search-announcement/
when sending this document
```json
{
  "id": 2,
  "title": "And Your Bird Can Sing",
  "_vectors": [
    [0.5, 3, 1],
    [-0.2, 4, 6],
  ],
},
```
you get code: invalid_vector_dimensions - type: invalid_request - link: https://docs.meilisearch.com/errors#invalid_vector_dimensions

if I change it to 
```json
{
  "id": 2,
  "title": "And Your Bird Can Sing",
  "_vectors":   [0.5, 3, 1],
},
```
it works fine

**Meilisearch version:**
v1.3.2

**Additional context**
Error was introduced in this commit https://github.com/meilisearch/meilisearch/commit/a7e0f0de89ef8b7a0e261dbaff4f0893cbd1f7d6

**Comment by Strift** [comment]
Hey, thanks for opening an issue.

I think the error lies more in the example than in Meilisearch's code here. I fixed the example in the article. 🙏 

I'll let the @meilisearch/engine-team close the issue.

**Comment by curquiza** [comment]
Thank you @ahmednfwela for the report and @Strift for the reactivity 🙏

**Comment by Kerollmops** [comment]
Hey @ahmednfwela 👋

Thank you very much for the bug report. It is indeed a bug on the engine side. The engine must accept multiple vectors by document and only restrict the length of them, not the sum of the length of the vectors. That's a mistake. We will prioritize this fix. I already have a failing test on my side.

**Comment by curquiza** [comment]
Fixed by #4039 
will be integrated into v1.3.3 released in the coming days
